crates/sui-graphql-rpc/src/types/checkpoint.rs

```diff
@@ -509,3 +539,45 @@ impl Loader<DigestKey> for Db {
             .collect())
     }
 }
+
+#[async_trait::async_trait]
+impl Loader<SeqNum> for Db {
+    type Value = StoredRawCheckpoint;
+    type Error = Error;
+
+    async fn load(&self, keys: &[SeqNum]) -> Result<HashMap<SeqNum, StoredRawCheckpoint>, Error> {
+        use raw_checkpoints::dsl;
+
+        let checkpoint_ids = keys
+            .iter()
+            .map(|key| key.sequence_number)
+            .collect::<Vec<_>>();
+
+        let raw_checkpoints: Vec<StoredRawCheckpoint> = self
+            .execute(move |conn| {
+                async move {
+                    conn.results(move || {
+                        dsl::raw_checkpoints
+                            .filter(dsl::sequence_number.eq_any(checkpoint_ids.iter().cloned()))
+                    })
+                    .await
+                }
+                .scope_boxed()
+            })
+            .await
+            .map_err(|e| Error::Internal(format!("Failed to fetch raw checkpoints: {e}")))?;
+
+        let raw_checkpoints: BTreeMap<_, _> = raw_checkpoints
+            .into_iter()
+            .map(|raw| (raw.sequence_number, raw))
+            .collect();
+
+        Ok(keys
+            .iter()
+            .filter_map(|key| {
+                let raw = raw_checkpoints.get(&key.sequence_number).cloned()?;
+                Some((*key, raw))
+            })
+            .collect())
```

Comment:
nit: In this case, you can probably convert the `raw_checkpoints` 
vector directly into the desired return type, (avoiding creating the intermediate map and then cloning from it), because you can construct the key you are fetching from the `StoredRawCheckpoint` that it points to.

Reply:
Absolutely, thanks for catching this!